lib/ansible/plugins/filter/core.py

```diff
@@ -99,8 +99,18 @@ def to_bool(a):
     return False
 
 
-def to_datetime(string, format="%Y-%m-%d %H:%M:%S"):
-    return datetime.datetime.strptime(string, format)
+def to_datetime(string, format="%Y-%m-%d %H:%M:%S", fail_on_error=False):
+    '''
+    Normal behaviour is to return datetime.
+    If toggle(fail_on_error) is True, in any value error it will fail.
+    '''
+    if fail_on_error:
+        try:
+            return datetime.datetime.strptime(string, format)
```

Comment:
Please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example. Also, add testcase for this change.

Reply:
Thanks a lot for the review, I added your suggestion and the changelog fragment . What kind of test is required in this case, any specific tests you could suggest me to check?